client/src/infographic/Infographic.js

```diff
@@ -351,17 +352,76 @@ class InfoGraph_ extends React.Component {
   }
 }
 
+const set_redirect_msg = (msg) => sessionStorage.setItem("redirected_msg", msg);
+const redirect_to_intro = (subject) => {
+  set_redirect_msg(
+    "You're redirected to intro because the data area is invalid"
+  );
+  window.location.replace(infograph_href_template(subject, "intro"));
+  window.location.reload();
+};
+
 const is_fake_infographic = (subject) =>
   !_.isUndefined(subject.is_fake) && subject.is_fake;
 const Infographic = ({
   match: {
     params: { level, subject_id, active_bubble_id, options },
   },
 }) => {
+  const is_level_valid = _.chain(Subject).keys().includes(level).value();
+  if (!is_level_valid) {
+    set_redirect_msg(
+      "The level of organization you tried to visit is invalid. We redirected you to homepage."
+    );
+    window.location.replace("#home");
+    window.location.reload();
+  }
   const SubjectModel = Subject[level];
   const subject = SubjectModel.lookup(subject_id);
   const bubble_id = bubble_defs[active_bubble_id] ? active_bubble_id : null;
 
+  if (!subject) {
+    const parent_dept_code = _.split(subject_id, "-")[0];
+    const parent_dept = Dept.lookup(parent_dept_code);
+    if ((level === "program" || level === "crso") && parent_dept) {
+      set_redirect_msg(
+        `The ${text_maker(
+          level
+        )} you tried to visit is either dead or invalid. We redirected you to its parent department.`
+      );
+      window.location.replace(`#orgs/dept/${parent_dept.id}/infograph/intro`);
+    } else {
+      set_redirect_msg(
+        "The organization you tried to visit is invalid. We redirected you to homepage."
+      );
+      window.location.replace("#home");
+    }
+    window.location.reload();
+  } else if (!bubble_id) {
+    redirect_to_intro(subject);
+  } else if (options) {
+    (() => {
```

Comment:
I think we can remove the anonymous function now and execute the try block directly?